sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ProcedureCatalog.java

```diff
@@ -34,4 +34,11 @@ public interface ProcedureCatalog extends CatalogPlugin {
    * @return the loaded unbound procedure
    */
   UnboundProcedure loadProcedure(Identifier ident);
+
+  /**
+   * List all procedures in the specified namespace.
+   *
+   * If a null or empty array is passed, all procedures are returned.
```

Comment:
Note: I put this convention for better UX to allow user to see all procedures in catalog, let me know thoughts @aokolnychyi @cloud-fan

Reply:
I am not sure how useful this would be given that there are always the default namespace and catalog. This method will be called by Spark and there would always be some default namespace. What do we do in similar interfaces?

Reply:
I guess a broader question:

Should 'show procedures' should return all in current catalog?  and 'show procedures in cat' should return all procedures in cat?

If so, we can have support at the catalog level, or else have Spark code loop through all namespaces and make multiple calls to the catalog?

Reply:
My understanding is that we have to resolve to the default catalog unless the catalog name is provided explicitly and list procedures in the currently default namespace. What do we do in `listFunctions` and `listViews`?

Reply:
OK they dont list beyond the current, I can follow that then